src/gsc_sql.c

```diff
@@ -0,0 +1,379 @@
+    /* Set the version, (a pointer of type char const), to the mysql client version */
+    const char *version = mysql_get_client_info();
+
+    /* Print out the MySQL Client Version for the user.
+     * This is an indication of whether or not cod4x is properly reading <mysql/mysql.h>
+     * I recommend using this as a basic test
+     */
+    Scr_AddString(version);
+    Com_Printf("MySQL Version: %s\n", version);
+    return;
+}
+
+/* =================================================================
+* URL
+    http://dev.mysql.com/doc/refman/5.7/en/mysql-init.html
+
+* Description
+    Allocates or initializes a MYSQL object suitable for mysql_real_connect().
+    If mysql is a NULL pointer, the function allocates, initializes, and returns
+    a new object. Otherwise, the object is initialized and the address of the
+    object is returned. If mysql_init() allocates a new object, it is freed
+    when mysql_close() is called to close the connection.
+
+* Return Value(s)
+    An initialized MYSQL* handle. NULL if there was insufficient memory to allocate a new object.
+   ================================================================= */
+void gsc_mysql_init() {
+
+    if (Scr_GetNumParam() != 0) {
+        Scr_Error("Usage: gsc_mysql_init();\n");
+    }
+
+    /* Print to let the user know that the function initalized */
+    Com_DPrintf("gsc_mysql_init()\n");
+
+    /* Sets my, (a pointer of type MYSQL), to the MySQL Connection Handle ID */
+    MYSQL *my = mysql_init(NULL);
+
+    if (my == NULL) {
+        Scr_AddUndefined();
+        Scr_Error("\nSomething went wrong when trying to get the MySQL CID\n");
+        return;
+    }
+
+    /* Returns the MySQL Handle Connection ID */
+    Scr_AddInt((int)my);
+}
+
+/* =================================================================
+* URL
+    http://dev.mysql.com/doc/refman/5.7/en/mysql-real-connect.html
+
+* Description
+    mysql_real_connect() attempts to establish a connection to a
+    MySQL database engine running on host. mysql_real_connect()
+    must complete successfully before you can execute any other
+    API functions that require a valid MYSQL connection handle structure.
+
+* Return Value(s)
+    A MYSQL* connection handle if the connection was successful, NULL
+    if the connection was unsuccessful. For a successful connection,
+    the return value is the same as the value of the first parameter.
+   ================================================================= */
+void gsc_mysql_real_connect() {
+
+    int mysql = Scr_GetInt(0);
+    char *host = Scr_GetString(1);
+    char *user = Scr_GetString(2);
+    char *pass = Scr_GetString(3);
+    char *db = Scr_GetString(4);
+    int port = Scr_GetInt(5);
+    MYSQL* mysql2;
+
+    /* On *Unix based systems, using "localhost" instead of 127.0.0.1 causes a unix socket error, we replace it*/
+    if (strcmp(host, "localhost") == 0) {
+        host = "127.0.0.1";
+    }
+
+    if (Scr_GetNumParam() != 6) {
```

Comment:
Is this because the parameters are passed implicitly?  I figured you couldn't check a value without getting it prior. I guess it makes sense to check beforehand.